reporter/src/main/kotlin/reporters/OpossumReporter.kt

```diff
@@ -0,0 +1,605 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Copyright (C) 2021 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import com.fasterxml.jackson.databind.json.JsonMapper
+
+import java.io.File
+import java.io.FileOutputStream
+import java.time.LocalDateTime
+import java.util.SortedMap
+import java.util.SortedSet
+import java.util.UUID
+import java.util.zip.Deflater
+
+import kotlin.math.min
+
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream
+import org.apache.commons.compress.compressors.gzip.GzipParameters
+
+import org.ossreviewtoolkit.model.CuratedPackage
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.utils.getPurlType
+import org.ossreviewtoolkit.model.utils.toPurl
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.spdx.SpdxCompoundExpression
+import org.ossreviewtoolkit.spdx.SpdxExpression
+import org.ossreviewtoolkit.spdx.SpdxLicense
+import org.ossreviewtoolkit.spdx.SpdxOperator
+import org.ossreviewtoolkit.spdx.getLicenseText
+import org.ossreviewtoolkit.utils.log
+
+fun resolvePath(left: String, right: String = "", isDirectory: Boolean? = null): String {
+    val isDir = isDirectory == true || (isDirectory == null && (if (right != "") right else left).last() == '/')
+    return "/$left/$right"
+        .replace(Regex("/*$"), if (isDir) "/" else "")
+        .replace(Regex("/+"), "/")
+        .ifEmpty { "/" }
+}
+
+fun resolvePath(pieces: List<String>): String {
```

Comment:
This function could be `internal` and turned into a [single-expression function](https://kotlinlang.org/docs/functions.html#single-expression-functions).